lib/wick-get-iface-ip

```diff
@@ -20,36 +20,58 @@
 #
 # Returns 0 on success, non-zero on failure.
 wickGetIfaceIp() {
-    local ip list
+    local list target interface allInterfaces
 
-    if [[ "$2" == '*' ]]; then
-        list=()
+    wickGetArgument target 0 "$@"
+    wickGetArgument interface 1 "$@"
 
-        while read ip; do
-            ip=${ip#*inet addr:}
-            ip=${ip#*inet }
-            ip=${ip%% *}
-            list[${#list[@]}]=$ip
-        done < <(ifconfig -v 2>/dev/null | grep 'inet ' || echo "")
+    list=()
+    allInterfaces=false
 
+    if [[ "$interface" == "*" ]]; then
+        interface=""
+        allInterfaces=true
+    fi
+
+    if wickCommandExists ip; then
+        cmd=(ip addr show)
+
+        if [[ -n "$interface" ]]; then
+            cmd[${#cmd[@]}]="$interface"
+        fi
+
+        list=( $("${cmd[@]}" 2>/dev/null | grep -Po "inet\s+\K[\d.]+") )
```

Comment:
Good catch.  It -P doesn't exist for bsd grep

5fce15c99f1d87bbb8177a92f6ed115f9cf4666d